src/main/java/com/treasuredata/client/TDHttpClient.java

```diff
@@ -118,64 +109,55 @@ protected SimpleDateFormat initialValue()
 
     public TDHttpClient(TDClientConfig config)
     {
         this.config = config;
-        this.httpClient = config.useSSL ? new HttpClient(new SslContextFactory()) : new HttpClient();
-        this.headers = config.headers;
-        httpClient.setConnectTimeout(config.connectTimeoutMillis);
-        httpClient.setIdleTimeout(config.idleTimeoutMillis);
-        httpClient.setTCPNoDelay(true);
-        QueuedThreadPool executor = new QueuedThreadPool(config.connectionPoolSize, 2);
-        executor.setDaemon(true);
-        executor.setName("td-client");
-        httpClient.setExecutor(executor);
-        httpClient.setScheduler(new ScheduledExecutorScheduler("td-client-scheduler", true));
-        httpClient.setCookieStore(new HttpCookieStore.Empty());
-        httpClient.setUserAgentField(new HttpField(HttpHeader.USER_AGENT, "td-client-java-" + TDClient.getVersion()));
-        if (config.requestBufferSize.isPresent()) {
-            httpClient.setRequestBufferSize(config.requestBufferSize.get());
-        }
-        if (config.responseBufferSize.isPresent()) {
-            httpClient.setResponseBufferSize(config.responseBufferSize.get());
-        }
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        builder.connectTimeout(config.connectTimeoutMillis, TimeUnit.MILLISECONDS);
+        builder.readTimeout(config.readTimeoutMillis, TimeUnit.MILLISECONDS);
+        CookieManager cookieManager = new CookieManager(null, CookiePolicy.ACCEPT_ORIGINAL_SERVER);
+        builder.cookieJar(new JavaNetCookieJar(cookieManager));
 
         // Proxy configuration
         if (config.proxy.isPresent()) {
             final ProxyConfig proxyConfig = config.proxy.get();
             logger.trace("proxy configuration: " + proxyConfig);
-            HttpProxy httpProxy = new HttpProxy(new Origin.Address(proxyConfig.getHost(), proxyConfig.getPort()), proxyConfig.useSSL());
+            // TODO Support https proxy
+            builder.proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyConfig.getHost(), proxyConfig.getPort())));
 
-            // Do not proxy requests for the proxy server
-            httpProxy.getExcludedAddresses().add(proxyConfig.getHost() + ":" + proxyConfig.getPort());
-            httpClient.getProxyConfiguration().getProxies().add(httpProxy);
             if (proxyConfig.requireAuthentication()) {
-                httpClient.getAuthenticationStore().addAuthenticationResult(new ProxyAuthResult(proxyConfig));
+                builder.proxyAuthenticator(new ProxyAuthenticator(proxyConfig));
             }
         }
+        // connection pool
+        ConnectionPool connectionPool = new ConnectionPool(config.connectionPoolSize, 5, TimeUnit.MINUTES);
+        builder.connectionPool(connectionPool);
+
+        // Build OkHttpClient
+        this.httpClient = builder.build();
+        this.headers = ImmutableMultimap.copyOf(config.headers);
+
+//        if (config.requestBufferSize.isPresent()) {
```

Comment:
It seems there is no configuration for the request/response buffer size. We should remove these configs